### ISSUE TITLE
[BUG]: getRepository 'typeorm' is deprecated

### ISSUE THREAD

**Issue by jrAtAustin** [nykula/connect-typeorm#37]
### Bug description

The documentation for connect-typeorm shows the session store connection being set to the entity repository from the depreciated TypeORM **getRepository** method. Is there a replacement for **getRepository** or a workaround for this issue?

### Steps to reproduce

See documentation (README.md) for this repository.

**Comment by jrAtAustin** [comment]
I found an alternative:
app.get(DataSource).getRepository(TypeORMSession)

**Comment by jrAtAustin** [comment]
Closing

**Comment by freshgiammi** [comment]
Hi, sorry for the late reply.
You're correct, **TypeORM changed how getRepository() is used since v0.3** (https://typeorm.io/data-source#creating-a-new-datasource) and I forgot to update the documentation. 

I'm reopening this issue as a memo to fix it, but a correct implementation would now be (this is how I set everything up in the index file):

```typescript
import { DataSource, DataSourceOptions } from 'typeorm';
import { TypeormStore } from 'connect-typeorm';
import { Session } from './entity/Session';
import express from 'express';
import ExpressSession from 'express-session';


// Define express variable
const app = express();

// Create an exportable `TypeormStore` to allow access everywhere
export const sessionStore = new TypeormStore({
    cleanupLimit: 2,
    limitSubquery: false, // If using MariaDB.
    ttl: 86400,
})

// Create an exportable `DataSource` to allow `getRepository()` access everywhere. 
// `databaseOptions` is an object of type DataSourceOptions and may depend on your configuration/needs.
export const dataSource = new DataSource(databaseOptions);

// Initialize DataSource
dataSource.initialize()
.then(async () => {

    const sessionRepository = dataSource.getRepository(Session);
    app.use(
        ExpressSession({
            resave: false,
            saveUninitialized: false,
            store: sessionStore.connect(sessionRepository),
            // Cookie settings can be added here based on your configuration/needs. 
            // See: https://expressjs.com/en/resources/middleware/session.html
            secret: 'keyboard cat' // This should not be hardcoded, an .env variable is preferred.
        })
    );

    /* Everything else required by your app: setting up routes, starting server on HTTP/S, etc... */

}).catch((error) => console.log(error));
```

**Comment by srini-leanfolks** [comment]
> Hi, sorry for the late reply. You're correct, **TypeORM changed how getRepository() is used since v0.3** (https://typeorm.io/data-source#creating-a-new-datasource) and I forgot to update the documentation.
> 
> I'm reopening this issue as a memo to fix it, but a correct implementation would now be (this is how I set everything up in the index file):
> 
> ```ts
> import { DataSource, DataSourceOptions } from 'typeorm';
> import { TypeormStore } from 'connect-typeorm';
> import { Session } from './entity/Session';
> import express from 'express';
> import ExpressSession from 'express-session';
> 
> 
> // Define express variable
> const app = express();
> 
> // Create an exportable `TypeormStore` to allow access everywhere
> export const sessionStore = new TypeormStore({
>     cleanupLimit: 2,
>     limitSubquery: false, // If using MariaDB.
>     ttl: 86400,
> })
> 
> // Create an exportable `DataSource` to allow `getRepository()` access everywhere. 
> // `databaseOptions` is an object of type DataSourceOptions and may depend on your configuration/needs.
> export const dataSource = new DataSource(databaseOptions);
> 
> // Initialize DataSource
> dataSource.initialize()
> .then(async () => {
> 
>     const sessionRepository = dataSource.getRepository(Session);
>     app.use(
>         ExpressSession({
>             resave: false,
>             saveUninitialized: false,
>             store: sessionStore.connect(sessionRepository),
>             // Cookie settings can be added here based on your configuration/needs. 
>             // See: https://expressjs.com/en/resources/middleware/session.html
>             secret: 'keyboard cat' // This should not be hardcoded, an .env variable is preferred.
>         })
>     );
> 
>     /* Everything else required by your app: setting up routes, starting server on HTTP/S, etc... */
> 
> }).catch((error) => console.log(error));
> ```

But it is not working with NestJS. If possible, please provide an example of using it in the NestJS

**Comment by freshgiammi** [comment]
> But it is not working with NestJS. If possible, please provide an example of using it in the NestJS

Yeah, that's an example configuration for an Express backend. I yet have to try running it in Nest. In the meanwhile, please refer to this https://docs.nestjs.com/techniques/session for how to set up a session middleware, it should point you in the right direction.

**Comment by rajamoulimallareddy** [comment]
> > But it is not working with NestJS. If possible, please provide an example of using it in the NestJS
> 
> Yeah, that's an example configuration for an Express backend. I yet have to try running it in Nest. In the meanwhile, please refer to this https://docs.nestjs.com/techniques/session for how to set up a session middleware, it should point you in the right direction.

tried this too its running fine with no issues but. data is not getting saved.

**Comment by freshgiammi** [comment]
> tried this too its running fine with no issues but. data is not getting saved.

I may have some time this week or next to look at it, could you please share a repo with the implementation?
 I'll need to reproduce it and I don't currently have anything running with Nest to try quickly, so a basic template with it would be nice.

**Comment by hayzedDev** [comment]
If you're using it with nest.js, you can do it this way

```
import { TypeormStore } from 'connect-typeorm';
import * as ExpressSession from 'express-session'; 
import { NestFactory } from '@nestjs/core';
import { Session } from './entities/session.entity';
import { DataSource } from 'typeorm';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);
  const sessionRepo = app.get(DataSource).getRepository(Session);
  app.use(
    ExpressSession({
      secret: process.env.SESSION_SECRET,
      resave: true,
      name: process.env.SESSION_NAME,
      saveUninitialized: true,
      cookie: { maxAge: Number(process.env.COOKIE_MAX_AGE) },
      store: new TypeormStore().connect(sessionRepo),
    }),
  );

  app.use(passport.initialize());
  app.use(passport.session());

  await app.listen(Number(process.env.NODE_PORT) || 3000);
}
bootstrap();
```
